### PR TITLE
Remove employee from department

### PR DESCRIPTION
**Pull request by kennethk-1201**
Command format: refd id/<employee-id> n/<department-name>

Features:
- Allows you to remove an employee from a given department
- Throws error if employee does not exist in department

Created relevant tests on command-level and parser-level

Others:
- Repackaged modules into department and employees.

### CODE REVIEW

File: src/test/java/seedu/sudohr/logic/parser/department/RemoveEmployeeFromDepartmentCommandParserTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    @Test
+    public void parse_fieldsMissing_failure() {
+        String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                RemoveEmployeeFromDepartmentCommand.MESSAGE_USAGE);
+
+        assertParseFailure(parser, DEPARTMENT_NAME_DESC_ENGINEERING, expectedMessage);
+        assertParseFailure(parser, ID_DESC_AMY, expectedMessage);
+        assertParseFailure(parser, "", expectedMessage);
+    }
+
```
Comment (4ndrelim) [inline src/test/java/seedu/sudohr/logic/parser/department/RemoveEmployeeFromDepartmentCommandParserTest.java:38]: can consider parsing additional fields (prefixes) too, should result in failure

File: src/test/java/seedu/sudohr/logic/parser/department/RemoveEmployeeFromDepartmentCommandParserTest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    @Test
+    public void parse_invalidField_failure() {
+        String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                RemoveEmployeeFromDepartmentCommand.MESSAGE_USAGE);
+
+        assertParseFailure(parser, INVALID_ID_DESC + DEPARTMENT_NAME_DESC_ENGINEERING,
```
Comment (4ndrelim) [inline src/test/java/seedu/sudohr/logic/parser/department/RemoveEmployeeFromDepartmentCommandParserTest.java:44]: another edge case is ID of 0 since we ban its use.

Will update in UG that 0 is reserved for something. On this note, should we ever implement anything that uses '0'? to make our story more coherent.

File: src/test/java/seedu/sudohr/logic/commands/department/RemoveEmployeeFromDepartmentCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+import seedu.sudohr.model.SudoHr;
+import seedu.sudohr.model.department.Department;
+import seedu.sudohr.model.department.DepartmentName;
+import seedu.sudohr.model.employee.Employee;
+import seedu.sudohr.model.employee.Id;
+import seedu.sudohr.testutil.TypicalDepartmentNames;
+import seedu.sudohr.testutil.TypicalEmployees;
+
+public class RemoveEmployeeFromDepartmentCommandTest {
+
```
Comment (4ndrelim) [inline src/test/java/seedu/sudohr/logic/commands/department/RemoveEmployeeFromDepartmentCommandTest.java:31]: what is the expected behaviour of removing non-existent employee from non-existent department? error msg for which is thrown first?

File: src/test/java/seedu/sudohr/logic/commands/department/RemoveEmployeeFromDepartmentCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+    @Test
+    public void execute_employeeAcceptedByDepartment_addSuccessful() throws CommandException {
+
+        RemoveEmployeeFromDepartmentCommandTest.ModelStubAcceptingEmployeeRemoved modelStub =
+                new RemoveEmployeeFromDepartmentCommandTest.ModelStubAcceptingEmployeeRemoved();
+        modelStub.addEmployee(TypicalEmployees.ALICE);
+        Department toAdd = new Department(TypicalDepartmentNames.DEPARTMENT_NAME_FIRST);
+        modelStub.addDepartment(toAdd);
+
+        new AddEmployeeToDepartmentCommand(new Id("101"),
```
Comment (4ndrelim) [inline src/test/java/seedu/sudohr/logic/commands/department/RemoveEmployeeFromDepartmentCommandTest.java:42]: rather than relying on this class, possible to directly add employee to department using any of the avaiable methods? Was thinking if we wanna keep the test function as 'pure' as possible.

**Review by 4ndrelim — APPROVED** [review]
LGTM Great job!

File: src/test/java/seedu/sudohr/logic/commands/department/RemoveEmployeeFromDepartmentCommandTest.java (hunk 3 above)
Comment (kennethk-1201) [reply src/test/java/seedu/sudohr/logic/commands/department/RemoveEmployeeFromDepartmentCommandTest.java:31]: employee exception will be thrown first

File: src/test/java/seedu/sudohr/logic/parser/department/RemoveEmployeeFromDepartmentCommandParserTest.java (hunk 2 above)
Comment (kennethk-1201) [reply src/test/java/seedu/sudohr/logic/parser/department/RemoveEmployeeFromDepartmentCommandParserTest.java:44]: i'll bring it up in our meeting

File: src/main/java/seedu/sudohr/logic/parser/department/RemoveEmployeeFromDepartmentCommandParser.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        DepartmentName name = ParserUtil.parseDepartmentName(argMultimap.getValue(PREFIX_DEPARTMENT_NAME).get());
+
+        return new RemoveEmployeeFromDepartmentCommand(id, name);
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
```
[the remark this answers is not in the document]
Comment (kennethk-1201) [reply src/main/java/seedu/sudohr/logic/parser/department/RemoveEmployeeFromDepartmentCommandParser.java:48]: ok, will raise an issue so we can one shot settle this in the future

File: src/test/java/seedu/sudohr/logic/commands/department/RemoveEmployeeFromDepartmentCommandTest.java (hunk 4 above)
Comment (kennethk-1201) [reply src/test/java/seedu/sudohr/logic/commands/department/RemoveEmployeeFromDepartmentCommandTest.java:42]: true, i'll look into it

File: src/main/java/seedu/sudohr/logic/commands/department/RemoveEmployeeFromDepartmentCommand.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+ */
+public class RemoveEmployeeFromDepartmentCommand extends Command {
+
+    public static final String COMMAND_WORD = "refd";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Removes an employee from an existing department.\n"
+            + "Parameters: EMPLOYEE_ID, DEPARTMENT_NAME "
+            + "[" + PREFIX_DEPARTMENT_NAME + "NAME] "
+            + "Example: " + COMMAND_WORD + " " + PREFIX_ID + " 100 "
+            + PREFIX_DEPARTMENT_NAME + "Software Engineering";
```
[the remark this answers is not in the document]
Comment (kennethk-1201) [reply src/main/java/seedu/sudohr/logic/commands/department/RemoveEmployeeFromDepartmentCommand.java:27]: ok i'll bring it up in the meeting, but i think likely it should be treated the same as Name in Employee
